site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/repository/SiteStore.java

```diff
@@ -39,19 +39,30 @@ public SiteStore(FhirRepository repository,
      */
     @Override
     public String saveEntity(Site entity) {
+        String orgId = "";
 
-        // TODO: Check if the Organization already exists.
+        //Check if the Organization already exists.
+        String orgIdFound = repository.findOrganizationByName(entity.getName());
```

Comment:
I think this is more appropriate to include in the validation layer as the right place to include logic checks. I see the store an down as more of a "database" like...

Reply:
This has been addressed